TerriaJS/terriajs
# Add option `focusWorkbenchItems` to `initialCamera` setting.

### What this PR does

Currently when the map has preloaded workbench items, there is no way to automatically focus the map on the workbench items. This PR adds a new option `focusWorkbenchItems` to `initialCamera` setting that will automatically zoom the map to show the workbench items.

If a subsequent init source or the share URL specifies a different `initialCamera` setting, that will take precedence and this option will be ignored.

Caveat: The current implementation focuses the first workbench item. This will be fixed to focus all workbench items when we refactor the zoom code path. [Issue here](https://github.com/TerriaJS/terriajs/issues/6807).

[Example init file](https://gist.githubusercontent.com/na9da/70dee24e06e994d340bfb3c1eb45140f/raw/b12543c00ea8dd12668fe5da4c7390b34e8c5498/focus-workbench-items.json).

### Test me

- Open this [CI link](http://ci.terria.io/focus-workbench-items/#https://gist.githubusercontent.com/na9da/2c78674c66903a932106983683411339/raw/c750222ba5bf77959fdf56cd8f23380128872e10/no-focus-workbench-items.json) for this branch using a config that does NOT enable `focusWorkbenchItems`
- Observe that the map loads the home camera view
- Now open this [CI link](http://ci.terria.io/focus-workbench-items/#https://gist.githubusercontent.com/na9da/70dee24e06e994d340bfb3c1eb45140f/raw/b12543c00ea8dd12668fe5da4c7390b34e8c5498/focus-workbench-items.json) with [focusWorkbenchItems](https://gist.github.com/na9da/70dee24e06e994d340bfb3c1eb45140f#file-focus-workbench-items-json-L3) enabled
- Observe that the map automatically zooms to the workbench dataset
- Ensure this works for both Cesium and Leaflet viewers

### Checklist
